src/meta/src/barrier/schedule.rs

```diff
@@ -128,15 +142,17 @@ impl<S: MetaStore> BarrierScheduler<S> {
                 finish_rx,
                 is_create_mv,
             });
-            scheduleds.push((
+            scheduleds.push(Scheduled {
                 command,
-                once(Notifier {
+                notifiers: once(Notifier {
                     collected: Some(collect_tx),
                     finished: Some(finish_tx),
+                    checkpoint: true,
                     ..Default::default()
                 })
                 .collect(),
-            ));
+                checkpoint: true,
```

Comment:
It seems unnecessary to always be `true` here. For commands like dropping materialized views, rescheduling, or `resume`, a normal barrier should be enough.

I still suggest implying the `is_checkpoint` in `Command` itself so that we can control it more preciously. For plain commands, we may use a bool flag to represent it.